plugin/trino-hive/src/main/java/io/trino/plugin/hive/cos/CosServiceConfigurationProvider.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.cos;
+
+import com.google.common.base.Splitter;
+import io.trino.hdfs.DynamicConfigurationProvider;
+import io.trino.hdfs.HdfsContext;
+import org.apache.hadoop.conf.Configuration;
+
+import javax.inject.Inject;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+
+import static io.trino.hdfs.DynamicConfigurationProvider.setCacheKey;
+import static io.trino.plugin.hive.s3.TrinoS3FileSystem.S3_ACCESS_KEY;
+import static io.trino.plugin.hive.s3.TrinoS3FileSystem.S3_ENDPOINT;
+import static io.trino.plugin.hive.s3.TrinoS3FileSystem.S3_SECRET_KEY;
+
+public class CosServiceConfigurationProvider
+        implements DynamicConfigurationProvider
+{
+    private final Map<String, ServiceConfig> services;
+
+    @Inject
+    public CosServiceConfigurationProvider(HiveCosServiceConfig config)
+    {
+        services = ServiceConfig.loadServiceConfigs(config.getServiceConfig());
+    }
+
+    @Override
+    public void updateConfiguration(Configuration configuration, HdfsContext context, URI uri)
+    {
+        if (!uri.getScheme().equals("cos")) {
```

Comment:
could the scheme be null? If yes, then it's better to check ``!"cos".equals(uri.getScheme())``

Reply:
Looks like the other few implementations are null safe. So doesn't hurt to have 👍